src/Simplify.cpp

```diff
@@ -1449,7 +1449,9 @@ class Simplify : public IRMutator {
         } else if (is_one(b)) {
             expr = a;
         } else if (equal(a, b) &&
-                   !is_zero(b)) {
+                   !is_zero(b) &&
+                   // (n/0)/(n/0) should not be simplified to 1
+                   !(div_a && is_zero(div_a->b))) {
```

Comment:
I couldn't remember how all of the combinations work out, so I wrote [a little program](https://gist.github.com/kgnk/c4ea25b640ccbe52851b0bee0a734d1a).